app/controllers/works_controller.rb

```diff
@@ -0,0 +1,68 @@
+class WorksController < ApplicationController
+  before_action :find_work, only: [:show, :edit, :update, :destroy]
+
+  def index
+    @categories = Work.categories
+  end
+
+  def show
+    if !@work
+      flash[:error] = "No work with that ID found."
+      redirect_to works_path
+    end
+  end
+
+  def new
+    @work = Work.new
+    @categories = Work.categories
+  end
+
+  def create
+    work = Work.new(work_params)
+
+    work.save
+
+    redirect_to works_path
+  end
+
+  def edit
+    if !@work
+      flash[:error] = "No work with that ID found."
+      redirect_to works_path
```

Comment:
This error checking is repeated many times in this controller. You've already written the `find_work` controller filter, could you do this work there?